aptos-move/framework/aptos-framework/sources/Coin.move

```diff
@@ -18,41 +18,51 @@ module AptosFramework::Coin {
     const ENO_MINT_CAPABILITY: u64 = 7;
     const EDESTRUCTION_OF_NONZERO_TOKEN: u64 = 8;
 
-    // Core data structures
+    /// Core data structures
 
-    // Represents a set amount of coin
+    /// Main structure representing a coin/token in an account's custody.
     struct Coin<phantom CoinType> has store {
+        /// Amount of coin this address has.
         value: u64,
     }
 
-    // Represents ownership of coin
+    /// A holder of a specific coin type. This is a wrapper that allows
+    /// adding more data such as events and can be thought of as a "wallet".
```

Comment:
My sense is to keep the comment concise and not contains too much personal interpretation. since one person's understanding of wallet may not help to make sense of the comments here. how about removing "This is a wrapper that allows adding more data such as events and can be thought of as a "wallet"."

Reply:
Agreed.

Maybe:

```A holder of a specific coin types and associated event handles. These are kept in a single resource to ensure locality of data.```

Reply:
Updated